compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -1164,8 +1178,7 @@ refsFromDataCons = \case
     LF.DataRecord fields -> foldMap (refsFromType . snd) fields
     LF.DataVariant cons -> foldMap (refsFromType . snd) cons
     LF.DataEnum _ -> mempty
-    -- TODO https://github.com/digital-asset/daml/issues/12051
-    LF.DataInterface -> error "interfaces are not implemented"
+    LF.DataInterface -> mempty
```

Comment:
I'm not 100% sure what the intention here is, perhaps we need to consider the types of the interface methods?

Reply:
Yes, I think we need to collect references from the interface methods. But this line is fine, the `DataCons` doesn't have references. Collecting refs from interface methods should be added in `modRootRefs`, I think.

Reply:
Maybe it's worth leaving this for now and revisiting it later.

Reply:
I don’t think we should add anything here. Interface methods are exposed via typeclass methods and we collect those types. We probably also want to collect references for the `HasMethod` instances we generate (if we don’t already do).

Reply:
No, I don't think I did it for the `HasMethod` instances. Since those have three arguments (interface, method name and method type), I guess it should be enough to collect references for each of those? then,

1.  interface -> taken care of by the datatype itself
2. method name -> it's a symbol so we can probably just ignore it for this?
3. method type -> add entry in `modRootRefs` for method types

does that make sense?

Reply:
I _think_ `convertType` might actually be sufficient for the type already. To test (manually is probably fine for now), I think you can do this:
1. Create dependency with two modules `Iface` and `Ty`. `Ty` defines a type that you use as the method result type in `Iface` (but nothing else from Iface references this).
2. Have a main project that depends on dependency as a data-dep.

If things work correctly the stub Daml file for Iface has an import for Ty and things work. If they don’t, it’s missing the import and fails to compile.

Reply:
The manual test did pass, but I think it's for a different reason, namely, the interface methods get translated to top level functions, so the pass for top level bindings takes care of building the refs.